app/main.py

```diff
@@ -1 +1,42 @@
-#  write your code here
+class Animal:
+    def __init__(self, name: str, appetite: int,
+                 is_hungry: bool = True) -> None:
+        self.name = name
+        self.appetite = appetite
+        self.is_hungry = is_hungry
+
+    def print_name(self) -> None:
+        print(f"Hello, I'm {self.name}")
+
+    def feed(self) -> int:
+        if self.is_hungry is True:
+            print(f"Eating {self.appetite} food points...")
+            self.is_hungry = False
+            return self.appetite
+        else:
+            return 0
+
+
+class Cat(Animal):
+    def __init__(self, name: str, is_hungry: bool = True,
+                 appetite: int = 3) -> None:
+        super().__init__(name, appetite, is_hungry)
+
+    @staticmethod
+    def catch_mouse() -> None:
+        print("The hunt began!")
+
+
+class Dog(Animal):
+    def __init__(self, name: str, is_hungry: bool = True,
+                 appetite: int = 7) -> None:
+        super().__init__(name, appetite, is_hungry)
+
+    @staticmethod
+    def bring_slippers() -> None:
+        print("The slippers delivered!")
+
+
+def feed_animals(list_of_animals: list) -> int:
+    return sum(animal.feed() or animal.appetite for
+               animal in list_of_animals if animal.is_hungry)
```

Comment:
The expression `animal.feed() or animal.appetite` is incorrect because `animal.feed()` will return the appetite value if the animal is hungry, and 0 otherwise. Therefore, using `or animal.appetite` will incorrectly add the appetite value even if the animal was not fed. You should simply sum the results of `animal.feed()` for each animal.